### PR TITLE
chore: add relative link lint for `lint:md-links`

### LINKED ISSUE #2141: Markdown validation not validating markdown link fragments
We have links to sections on a page `#typegen` (see [example](https://github.com/FuelLabs/fuels-ts/blob/c9597f91c05e4f2320958d4de33e8f26652cff57/apps/docs/src/guide/fuels/commands.md?plain=1#L70-L71)). There is currently no validation to check that they exist and therefore, are broken.|

We should add validation to ensure that, all our links work as expects, so the user can navigate the documentation with ease.

### PR DESCRIPTION
**Pull request by petertonysmith94**
Closes #2141

---

- Added linting for markdown anchor links (#example)
- Moved CI linting to a separate workflow ([`code-lint`](https://github.com/FuelLabs/fuels-ts/blob/ps/fix/linting-markdown-links/.github/workflows/code-lint.yaml)) and removed `markdown-link-check` workflow.

### CODE REVIEW

File: .textlintrc.json [hunk 1]
```diff
@@ -2,12 +2,15 @@
...
+    "@elasticpath/no-dead-relative-link": true,
     "no-dead-link": {
       "ignoreRedirects": true,
       "checkRelative": false,
       "ignore": [
         "https://learnmeabitcoin.com/**",
-        "https://faucet-beta-5.fuel.network/"
+        "https://faucet-beta-5.fuel.network/",
+        "https://thebitcoinmanual.com/",
+        "https://forum.fuel.network/"
```
Comment (arboleya) [inline .textlintrc.json:13]: Did you experience lots of failures with the faucet and forum URLs?

**Review by danielbate — APPROVED** [review]

File: .knip.json [hunk 2]
```diff
@@ -23,8 +23,9 @@
     "open",
     "textlint",
     "textlint-rule-no-dead-link",
+    "@elasticpath/textlint-rule-no-dead-relative-link",
     "tree-kill",
     "ts-generator",
-    "webdriverio"
+    "webdriverio",
```
Comment (maschad) [inline .knip.json:29]: unnecessary
```suggestion
    "webdriverio"
```

**Review by maschad — APPROVED** [review]
minor suggestion but lgtm

**Review by danielbate — DISMISSED** [review]

**Review by danielbate — APPROVED** [review]

**Review by maschad — DISMISSED** [review]
Approving this but we should do some follow-up work in https://github.com/FuelLabs/fuels-ts/issues/1835#issuecomment-2108547538

**Review by maschad — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Pull request merged**